metricbeat/metricbeat.reference.yml

```
@@ -465,20 +465,32 @@ metricbeat.modules:
     - event
 
 # Kubernetes API server
+# (when running metricbeat as a pod)
 - module: kubernetes
   enabled: true
   metricsets:
     - apiserver
   hosts: ["https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}"]
 
 # Kubernetes proxy server
+# (when running metricbeat locally at hosts or as a daemonset + host network)
 - module: kubernetes
   enabled: true
   metricsets:
     - proxy
   hosts: ["localhost:10249"]
   period: 10s
 
+# Kubernetes controller manager
+# (URL and deployment will need to be modified to match the controller manager deployment)
+- module: kubernetes
+  enabled: true
+  metricsets:
+    - controllermanager
+  hosts: ["http://controller-manager:10252"]
```

I'd leave it with localhost as we use to do, if deployed in master nodes in the host network it should work.

```suggestion
  hosts: ["http://localhost:10252"]
```